tst/EnergyPlus/unit/SingleDuct.unit.cc

```diff
@@ -2835,3 +2835,183 @@ TEST_F(EnergyPlusFixture, setATMixerSizingProperties_Test)
+        "    ZoneHVAC:AirDistributionUnit,  !- Zone Equipment 1 Object Type",
+        "    Zone 1 ADU,            !- Zone Equipment 1 Name",
+        "    1,                       !- Zone Equipment 1 Cooling Sequence",
+        "    1;                       !- Zone Equipment 1 Heating or No-Load Sequence",
+        "ZoneHVAC:AirDistributionUnit,",
+        "    Zone 1 ADU,    !- Name",
+        "    Zone 1 Supply Inlet,     !- Air Distribution Unit Outlet Node Name",
+        "    AirTerminal:SingleDuct:ConstantVolume:Reheat,  !- Air Terminal Object Type",
+        "    Zone 1 TU Reheat;        !- Air Terminal Name",
+        "AirTerminal:SingleDuct:ConstantVolume:Reheat,",
+        "    Zone 1 TU Reheat,        !- Name",
+        "    ,                        !- Availability Schedule Name",
+        "    Zone 1 Supply Inlet,     !- Air Outlet Node Name",
+        "    Zone 1 TU Air Inlet,     !- Air Inlet Node Name",
+        "    1.0,                     !- Maximum Air Flow Rate {m3/s}",
+        "    Coil:Heating:Electric,   !- Reheat Coil Object Type",
+        "    Zone 1 Reheat Coil;      !- Reheat Coil Name",
+        "Coil:Heating:Electric,",
+        "    Zone 1 Reheat Coil,      !- Name",
+        "    ,                        !- Availability Schedule Name",
+        "    1,                       !- Efficiency",
+        "    Autosize,                !- Nominal Capacity of the Coil {W}",
+        "    Zone 1 VAV Reheat Coil Air Inlet,  !- Air Inlet Node Name",
+        "    Zone 1 Supply Inlet,     !- Air Outlet Node Name",
+        "    ;                        !- Temperature Setpoint Node Name",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool ErrorsFound = false;
+    HeatBalanceManager::GetZoneData(*state, ErrorsFound);
+    ASSERT_FALSE(ErrorsFound);
+    DataZoneEquipment::GetZoneEquipmentData(*state);
+    state->dataSize->TermUnitFinalZoneSizing.allocate(1);
+    state->dataSize->TermUnitSizing.allocate(1);
+    state->dataSize->FinalZoneSizing.allocate(1);
+    ZoneAirLoopEquipmentManager::GetZoneAirLoopEquipment(*state);
+    SingleDuct::GetSysInput(*state);
+    EXPECT_TRUE(compare_err_stream(""));
+
+    int SysNum = 1;
+
+    // test air flow sizing of TU with hard sized input for Maximum Air Flow Rate {m3/s}
+    state->dataSize->ZoneSizingRunDone = true;
+    state->dataSize->CurZoneEqNum = 1;
+    state->dataSize->CurTermUnitSizingNum = 1;
+    state->dataSize->TermUnitFinalZoneSizing(1).NonAirSysDesHeatVolFlow = 0.5;
+    state->dataSingleDuct->sd_airterminal(SysNum).SizeSys(*state);
+    // constant volume TUs should have ZoneMinAirFracDes = 1 (now set correctly in GetSysInput)
+    EXPECT_EQ(0.0, state->dataSingleDuct->sd_airterminal(SysNum).ZoneMinAirFracDes);
+    // if ZoneMinAirFracDes = 0, TermUnitSizing(1).AirVolFlow would = 0.5 (NonAirSysDesHeatVolFlow)
+    // since ZoneMinAirFracDes = 1, TermUnitSizing(1).AirVolFlow = hard sized value (i.e., 1.0)
```

Comment:
I missed these comments on the last pass and these are no longer true. I still don't like that ZoneMinAirFracDes = 0 for CV TUs but at this point I guess "ZoneMinAirFracDes is not used for CV TUs" is where we are at. I am now thinking about these unit tests and whether they provide fool proof checks going forward. Lots of work went into 1 line of code but worth it.